src/Core/Appplication/Service/ReportGeneratorFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Core\Appplication\Service;
+
+use SprykerSdk\Sdk\Core\Appplication\Service\Violation\ViolationReportGenerator;
+use SprykerSdk\SdkContracts\Entity\ContextInterface;
+use SprykerSdk\SdkContracts\Report\ReportGeneratorFactoryInterface;
+use SprykerSdk\SdkContracts\Report\ReportGeneratorInterface;
+
+class ReportGeneratorFactory implements ReportGeneratorFactoryInterface
+{
+    /**
+     * @var \SprykerSdk\Sdk\Core\Appplication\Service\Violation\ViolationReportGenerator
+     */
+    private ViolationReportGenerator $violationReportGenerator;
+
+    /**
+     * @param \SprykerSdk\Sdk\Core\Appplication\Service\Violation\ViolationReportGenerator $violationReportGenerator
+     */
+    public function __construct(ViolationReportGenerator $violationReportGenerator)
```

Comment:
As far as I understand, this class will be changed upon introducing new report type because of explicit ViolationReportGenerator type mentioned in the constructor, right? Can it be done in a way that will not require a such changes? Can we, for example, pass a tagged iterator with all report generators here and then resolve needed generator by get* method's parameters?